index.bs

```diff
@@ -3460,6 +3625,9 @@ To <dfn>find matching sources</dfn> given an [=attribution trigger=] |trigger|:
       * |a|'s [=attribution source/priority=] is less than |b|'s [=attribution source/priority=].
       * |a|'s [=attribution source/priority=] is equal to |b|'s [=attribution source/priority=] and |a|'s
          [=attribution source/source time=] is less than |b|'s [=attribution source/source time=].
+      * the result of running [=check if an attribution source and attribution trigger have matching attribution scopes=] with |a| and |trigger| is false
```

Comment:
Could someone please help me understand why we do all this -- add conditions to sorting and have an additional check after the sort -- rather than a simple condition like we have above:
```
If |source|'s [=attribution source/attribution destinations=] does not [=set/contains|contain=] |trigger|'s [=attribution trigger/attribution destination=], [=iteration/continue=].
```

Reply:
The source deactivation will happen across scopes. If we only return the selected the sources with matching scope, we will be performing deactivation within the scope. The correct logic is to deactivate all other losing sources even if there is no matching scope.

Reply:
In that case, I'm not sure this matches the explainer -- the explainer says sources are "deleted" across scopes only if an event report is created.

Reply:
Yes, that's the correct behavior. that's why we check if there is no matching scope, we return empty list. that's why we need the extra check after sorting.

Reply:
The explainer says sources are "deleted" across scopes only if an event report is created. https://github.com/WICG/attribution-reporting-api/blob/main/attribution_scopes.md?plain=1#L161

But spec deactivates as soon as top-level filters match.